### ISSUE TITLE
ImageHDU.scale fails with bad operand type for unary -

### ISSUE THREAD

**Issue by ivvv** [astropy/astropy#6399]
I am trying to scale an image and it fails with an error.  I use astropy v2.0
`astropy.__version__ = 2.0, numpy.__version__ = 1.13.1`

Here is a simple code:

```python
import numpy as np
from astropy.io import fits

hdu1 = fits.PrimaryHDU()
hdu2 = fits.ImageHDU(np.random.rand(100,100))
hdu2.scale(type='uint8',bscale=1,bzero=0)
```

```python
     np.add(self.data, -_zero, out=self.data, casting='unsafe')
 TypeError: bad operand type for unary -: 'NoneType'
```

I tried to set `hdu2.header['BSCALE']` and `'BZERO'` but I see no change in behaviour.

I see a comment in the code, regarding this putative `_zero`, but apparently this is broken or I am doing something wrong.

Thanks for looking into this.

Best regards,
Ivan

**Comment by saimn** [comment]
It seems it could be related to #5053 which added [this](https://github.com/astropy/astropy/blob/500700cce3baeeb8e746b101a4cc5a2b1692721b/astropy/io/fits/hdu/image.py#L171), which set `_orig_bzero` and `_orig_bscale` to None in our case. ping @mwcraig as you added this code :wink:

**Comment by saimn** [comment]
And #5955 tries to fix exactly that it seems.

**Comment by ivvv** [comment]
Following the comments I managed to find a workaround/cheat:

`hdu2.scale(type='uint8',bscale=1.0+1.0e-8,bzero=1.0e-8)
`

this one seems to work, but it is a bug and has to be fixed.

**Comment by bsipocz** [comment]
@saimn - If you have time to review #5955, and thinks it's ready, it can still go into 2.0.1.

**Comment by saimn** [comment]
Fixed by #5955
